CondTools/SiPhase2Tracker/plugins/DTCCablingMapProducer.cc

```diff
@@ -178,30 +195,41 @@ void DTCCablingMapProducer::LoadModulesToDTCCablingMapFromCSV(
           uint32_t detIdRaw;
 
           try {
-            detIdRaw = std::stoi(csvColumn[csvFormat_idetid_]);
+            detIdRaw = std::stoi(csvColumn.at(csvFormat_idetid_));
           } catch (std::exception const& e) {
             if (verbosity_ >= 0) {
               edm::LogError("CSVParser") << "-- malformed DetId string in CSV file: \"" << csvLine << "\"" << endl;
             }
             throw e;
           }
 
-          unsigned const dtc_id = strtoul(csvColumn[csvFormat_idtcid_].c_str(), nullptr, 10);
+          unsigned const dtc_id = strtoul(csvColumn.at(csvFormat_idtcid_).c_str(), nullptr, 10);
           unsigned gbt_id;
           unsigned elink_id;
 
-          if (generate_fake_valid_gbtlink_and_elinkid_) {
-            for (gbt_id = gbt_id_minvalue; gbt_id < gbt_id_maxvalue + 1u; ++gbt_id) {
+          switch (dummy_fill_mode_) {
+            default:
+            case DUMMY_FILL_DISABLED:
+              gbt_id = strtoul(csvColumn.at(csvFormat_igbtlinkid_).c_str(), nullptr, 10);
+              elink_id = strtoul(csvColumn.at(csvFormat_ielinkid_).c_str(), nullptr, 10);
+              break;
+            case DUMMY_FILL_ELINK_ID:
+              gbt_id = strtoul(csvColumn.at(csvFormat_igbtlinkid_).c_str(), nullptr, 10);
               for (elink_id = elink_id_minvalue; elink_id < elink_id_maxvalue + 1u; ++elink_id) {
```

Comment:
Just having a quick diagonal look.

Sorry am not sure I understand this on the e-link Id: 
elink_id_minvalue = 0
elink_id_maxvalue = 7
So here you are looping from 0 to 7 included, while there are always 7 ports max only on a GBT (right?).
Why not:
elink_id = elink_id_minvalue; elink_id < elink_id_maxvalue; ++elink_id)
only?

Reply:
Good, you caught it. I will change it!